nav2_smac_planner/src/a_star.cpp

```diff
@@ -189,9 +195,13 @@ void AStarAlgorithm<Node2D>::setGoal(
   if (dim_3 != 0) {
     throw std::runtime_error("Node type Node2D cannot be given non-zero goal dim 3.");
   }
+  _goals.clear();
+  _goals_coordinates.clear();
+  _goalsSet.clear();
```

Comment:
Do we really need both the set and the vector? Can we remove the vector and use the set for all?